src/base/StringView.h

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (C) 1996-2019 The Squid Software Foundation and contributors
+ *
+ * Squid software is distributed under GPLv2+ license and includes
+ * contributions from numerous individuals and organizations.
+ * Please see the COPYING and CONTRIBUTORS string_views for details.
+ */
+
+#ifndef SQUID_BASE_STRING_VIEW_H
+#define SQUID_BASE_STRING_VIEW_H
+
+#include <iosfwd>
+
+// TODO: Replace with std::string_view after switching to C+17
+/// a reference to portion of a character sequence
+/// 0-termination is neither assumed nor honored -- NUL is not treated specially
+class StringView
+{
+public:
+    StringView(): start_(nullptr), size_(0) {}
+    StringView(const char * const start, const size_t len): start_(start), size_(len) {}
```

Comment:
There is no protection here against StringView(nullptr, 1) which result is memory access violations on all the strncmp, memcmp and anything protected only by empty().

Reply:
Done: `assert(start || !len)` added in branch commit c363dbf.

Please note that this addition introduces an inconsistency with `std::string_view` which might bite us after we migrate to C++17: Unreported assertions would become nil pointer dereferences (i.e. undefined behavior). The migration may also trigger new Coverity issues but that is even less important.